File: src/main/java/com/team3/gudit/sale/dto/response/SaleStatusUpdateResponseDto.java
package com.team3.gudit.sale.dto.response;

import com.team3.gudit.sale.domain.entity.Sale;
import com.team3.gudit.sale.domain.enums.SaleStatus;

public record SaleStatusUpdateResponseDto(
        Long saleId,
        SaleStatus status
) {
    public static SaleStatusUpdateResponseDto from(Sale sale) {
        return new SaleStatusUpdateResponseDto(
                sale.getId(),
                sale.getStatus()
        );
    }
}

File: src/main/kotlin/com/team3/gudit/sale/dto/response/SaleStatusUpdateResponseDto.kt
package com.team3.gudit.sale.dto.response

import com.team3.gudit.sale.domain.entity.Sale
import com.team3.gudit.sale.domain.enums.SaleStatus

@JvmRecord
data class SaleStatusUpdateResponseDto(
    val saleId: Long?,
    val status: SaleStatus?,
) {
    companion object {
        @JvmStatic
        fun from(sale: Sale): SaleStatusUpdateResponseDto =
            SaleStatusUpdateResponseDto(
                sale.id,
                sale.status,
            )
    }
}
